src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4344,7 +4377,7 @@ public override BoundNode VisitLocalFunctionStatement(BoundLocalFunctionStatemen
                     useMethodSignatureParameterTypes: false,
                     methodSignatureOpt: null,
                     returnTypes: null,
-                    initialState: GetVariableState(),
+                    initialState: GetVariableState(this.TopState()),
```

Comment:
>GetVariableState(this.TopState()), [](start = 34, length = 34)

loosely related to this PR: I don't remember what we had decided for local functions. Should they be analyzed at point of use like lambdas?
I think we do that for definite assignment, but I don't remember if there's a reason we couldn't also do it for nullable analysis.

Reply:
@agocke is the person to ask about that.  I think we want to do something like we do for definite assignment analysis: perform an analysis that is equivalent to inlining them at the point of use.

---
In reply to: [267591240](https://github.com/dotnet/roslyn/pull/34300#discussion_r267591240) [](ancestors = 267591240)

Reply:
Started email thread

---
In reply to: [267977533](https://github.com/dotnet/roslyn/pull/34300#discussion_r267977533) [](ancestors = 267977533,267591240)